### DISCUSSION TITLE
Flask app freezes

### DISCUSSION THREAD

**Discussion by ramanraja** [pallets/flask#4388] [Q&A]
I have a Flask application with a gevent server and an Nginx reverse proxy.  (Additional info: the app runs on a Windows EC2 instance and uses a MySQL database).  This is my first Flask app and it works well, that is, most of the times.
But the application often freezes and the browser clients time out.  The only remedy is to press CTRL+C on the application console. When I do that, all the pending requests arrive in a flood, as if some queue has burst open.
My questions are:  (1) Under what circumstances is this possible ?  (2) More importantly, how to debug the main loop - Where to insert log events or probe the internals of the app, so that I can locate where exactly the requests are held up?

**Comment by ThiefMaster** [comment]
I think without you showing code (as minimal as possible to reproduce your problem) it's unlikely that someone can help you...

**Reply by ramanraja** [reply]
I agree that a minimal reproducible code is necessary, and I am trying to strip down my code to that point. (It has grown organically over several months). But my idea was to learn more about debugging Flask applications, and inspect the runtime behavior of the message pump. 
It does not throw any exceptions; so I was wondering if I can trace an incoming call till the time it hits my route function, where it gets queued up etc.  Thank you.

**Reply by ramanraja** [reply]
Update: I set `debug=True` in `socketio.run()`.  Effects:
1. My application froze
2. It did not respond to repeated pressing of CTRL+C
3. Browser clients kept indefinitely waiting for response. They did not even time out!
I am trying to dig into the code and understand what all happens *before* my first app route is called. 
I am not familiar with Flask code, so any help would be appreciated.
